### PR TITLE
Auto generate test PEM files at runtime using BouncyCastle

### LINKED ISSUE #6826: Auto-generate test PEM files at runtime for OpenSearch sink mTLS tests
Description                                                                                                                                       
                                                                                                                                                    
The mTLS client certificate tests added in #6755 commit static PEM files to the test resources directory (`test-client-cert.pem`, `test-client-key.pem`, `test-client-ca.pem`, `test-client-ca-key.pem`). Per review feedback from @sb2k16  and @dlvenable , these should be generated programmatically at test runtime using BouncyCastle, following the pattern already established in `ConnectionConfiguration_ServerTest.createTrustStoreFromPem()`.

What needs to change                                                                                                                              
   
- Remove the committed PEM files from data-prepper-plugins/opensearch/src/test/resources/                                                         
- Generate client CA, client certificate, and client key at test startup using BouncyCastle JcaX509CertificateConverter and JcaPKCS8Generator
- Update ConnectionConfiguration_ServerTest.ClientCertificateConfiguration and ConnectionConfigurationTests to use the generated certs            
- Update the integration test OpenSearchClientCertIT to generate certs in its @BeforeAll setup                                                    
                                                                                                                                                    
Context                                                                                                                                           

This was identified during the review of #6755. The PR was merged with the static PEM files to unblock the feature, with agreement to follow up.  
                                                            
References

- PR: #6755                                                                                                                                       
- Review comment: https://github.com/opensearch-project/data-prepper/pull/6755#issuecomment-4383517866

### PR DESCRIPTION
**Pull request by srikanthpadakanti**
### Description
Replace static PEM test files with runtime generated certificates using `BouncyCastle` for the OpenSearch sink mTLS tests. The test CA is now generated with proper BasicConstraints and KeyUsage extensions. Added edge case tests for wrong CA rejection and inline PEM content support.
 
### Issues Resolved
Resolves #6826 
https://github.com/opensearch-project/data-prepper/issues/6826                                                                                      
 
### Check List
- [ X ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
- [ X ] New functionality has javadoc added
- [ X ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfigurationTests.java [hunk 1]
```diff
@@ -66,10 +67,20 @@ class ConnectionConfigurationTests {
...
+    private static String TEST_CLIENT_CERT_PATH;
+    private static String TEST_CLIENT_KEY_PATH;
+
+    @BeforeAll
+    static void generateClientCertificates() throws Exception {
+        final TestCertificateGenerator.GeneratedCertificateAuthority ca =
+                TestCertificateGenerator.generateClientCertificateAuthority();
+        final TestCertificateGenerator.GeneratedCertificate clientCert =
+                TestCertificateGenerator.generateClientCertificate(ca.getCertificate(), ca.getPrivateKey());
+        TEST_CLIENT_CERT_PATH = TestCertificateGenerator.writePemToTempFile(
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfigurationTests.java:83]: We should use either JUnit's `@TempDir` or have an `@AfterAll` to clean up the files.

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thanks @srikanthpadakanti !

File: data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfigurationTests.java (hunk 1 above)
Comment (srikanthpadakanti) [reply data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfigurationTests.java:83]: Ack.

**Comment by srikanthpadakanti** [comment]
@dlvenable AWS secrets test is unrelated to this change. 

main has a compile error in OpenSearchSink.java where #6771 references a constructor-local variable from `doInitializeInternal()` after #6795 refactored it out of scope.

**Review by dlvenable — APPROVED** [review]
